sdk/appconfiguration/azure-data-appconfiguration/src/configuration_client.cpp

```diff
@@ -0,0 +1,2316 @@
+}
+
+Azure::Response<GetKeyValueResult> ConfigurationClient::GetKeyValue(
+    std::string const& key,
+    std::string const& accept,
+    GetKeyValueOptions const& options,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath("kv/");
+  url.AppendPath(!key.empty() ? Core::Url::Encode(key) : "null");
+
+  url.AppendQueryParameter("api-version", Core::Url::Encode(m_apiVersion));
+  url.AppendQueryParameter("label", Core::Url::Encode(options.Label));
+  {
+    std::string valueStr;
+    for (size_t i = 0; i < options.Select.size(); ++i)
+    {
+      if (i != 0)
+      {
+        valueStr += ',';
+      }
+
+      valueStr += Core::Url::Encode(options.Select[i].ToString());
+    }
+
+    url.AppendQueryParameter("$Select", valueStr);
+  }
+
+  Core::Http::Request request(Core::Http::HttpMethod::Get, url);
+
+  request.SetHeader("Accept", accept);
+  request.SetHeader("Sync-Token", options.SyncToken);
+  request.SetHeader("Accept-Datetime", options.AcceptDatetime);
+  request.SetHeader("If-Match", options.IfMatch);
+  request.SetHeader("If-None-Match", options.IfNoneMatch);
+  request.SetHeader("x-ms-client-request-id", options.XMsClientRequestId.ToString());
+  auto rawResponse = m_pipeline->Send(request, context);
+  auto const httpStatusCode = rawResponse->GetStatusCode();
+
+  if (httpStatusCode != Core::Http::HttpStatusCode::Ok)
+  {
+    throw Core::RequestFailedException(rawResponse);
+  }
+
+  GetKeyValueResult response{};
+  {
+    auto const& responseBody = rawResponse->GetBody();
+    if (responseBody.size() > 0)
+    {
+      auto const jsonRoot
+          = Core::Json::_internal::json::parse(responseBody.begin(), responseBody.end());
+
+      response.Key = jsonRoot["key"].get<std::string>();
+
+      if (jsonRoot.contains("label") && !jsonRoot["label"].is_null())
+      {
+        response.Label = jsonRoot["label"].get<std::string>();
+      }
+
+      if (jsonRoot.contains("content_type") && !jsonRoot["content_type"].is_null())
+      {
+        response.ContentType = jsonRoot["content_type"].get<std::string>();
+      }
+
+      if (jsonRoot.contains("value") && !jsonRoot["value"].is_null())
+      {
+        response.Value = jsonRoot["value"].get<std::string>();
+      }
+
+      if (jsonRoot.contains("last_modified") && !jsonRoot["last_modified"].is_null())
+      {
+        response.LastModified = jsonRoot["last_modified"].get<std::string>();
+      }
+
+      if (jsonRoot.contains("tags"))
+      {
+        response.Tags = std::map<std::string, std::string>{};
+
+        for (auto const& kv : jsonRoot["tags"].items())
+        {
+          std::string value{};
+          value = kv.value().get<std::string>();
+          response.Tags.Value().emplace(kv.key(), value);
+        }
+      }
+
+      if (jsonRoot.contains("locked") && !jsonRoot["locked"].is_null())
+      {
+        response.Locked = jsonRoot["locked"].is_string()
+            ? (jsonRoot["locked"].get<std::string>() == "true")
```

Comment:
Does the server really return a stringized boolean here rather than an actual boolean?

If you're making this accommodation for `Locked`, why not other fields where there might be type confusion issues?

Reply:
Good point. I don't believe other language SDKs like GoLang have this string/bool flexibility when it comes to deserializing the JSON response into the model:
https://github.com/Azure/azure-sdk-for-go/blob/37b7f07b4b82d5bd5b5c6b89a0ec0820003dfeee/sdk/data/azappconfig/internal/generated/zz_models_serde.go#L227-L229

This flexibility is introduced in the emitter. Not sure why.

Reply:
https://github.com/Azure/autorest.cpp/issues/431